bokeh/properties.py

```diff
@@ -332,7 +389,14 @@ def __init__(self, delegate, help="", use_prefix=True):
 """
 
 class MetaHasProps(type):
-    def __new__(cls, class_name, bases, class_dict):
+    def __new__(meta_cls, class_name, bases, class_dict):
+        # HACK for now - we get called twice for some weird reason.
+        # __new__ used to be idempotent before PropertyDescriptor
+        # had to be transformed into Property, so it was OK, but now
+        # it isn't ... I think. Anyway, on the second call just do this?
+        # we need to figure this out for real and clean it up.
+        if '__properties__' in class_dict:
+            return type.__new__(meta_cls, class_name, bases, class_dict)
```

Comment:
Confirmed this works on py3. I did not realize `future` was a third party lib. Maybe we can just copy the definition of `with_metaclass` into our own tree, it's very short:

http://python-future.org/_modules/future/utils.html

Here's a cleaned up diff
```
diff --git a/bokeh/model.py b/bokeh/model.py
index aafd94c..ea3462d 100644
--- a/bokeh/model.py
+++ b/bokeh/model.py
@@ -3,7 +3,9 @@ from __future__ import absolute_import, print_function
 import logging
 logger = logging.getLogger(__file__)
 
-from six import add_metaclass, iteritems
+from future.utils import with_metaclass
+
+from six import iteritems
 
 from .properties import Any, HasProps, List, MetaHasProps, String
 from .query import find
@@ -64,8 +66,7 @@ class Viewable(MetaHasProps):
         else:
             raise KeyError("View model name '%s' not found" % view_model_name)
 
-@add_metaclass(Viewable)
-class Model(HasProps, CallbackManager):
+class Model(with_metaclass(Viewable, HasProps, CallbackManager)):
     """ Base class for all plot-related objects """
 
     name = String()
diff --git a/bokeh/properties.py b/bokeh/properties.py
index 611180c..6a07ff3 100644
--- a/bokeh/properties.py
+++ b/bokeh/properties.py
@@ -41,21 +41,23 @@ to be understood by BokehJS.
 """
 from __future__ import absolute_import, print_function
 
-import re
-import types
-import difflib
+import logging
+logger = logging.getLogger(__name__)
+
+import collections
+from copy import copy
 import datetime
 import dateutil.parser
-import collections
+import difflib
+from future.utils import with_metaclass
 from importlib import import_module
-from copy import copy
-from warnings import warn
 import inspect
-import logging
 import numbers
-logger = logging.getLogger(__name__)
+import re
+import types
+from warnings import warn
 
-from six import string_types, add_metaclass, iteritems
+from six import string_types, iteritems
 
 from . import enums
 from .util.string import nice_join
@@ -390,13 +392,6 @@ _EXAMPLE_TEMPLATE = """
 
 class MetaHasProps(type):
     def __new__(meta_cls, class_name, bases, class_dict):
-        # HACK for now - we get called twice for some weird reason.
-        # __new__ used to be idempotent before PropertyDescriptor
-        # had to be transformed into Property, so it was OK, but now
-        # it isn't ... I think. Anyway, on the second call just do this?
-        # we need to figure this out for real and clean it up.
-        if '__properties__' in class_dict:
-            return type.__new__(meta_cls, class_name, bases, class_dict)
         names = set()
         names_with_refs = set()
         container_names = set()
@@ -487,7 +482,7 @@ class MetaHasProps(type):
             path = class_dict["__example__"]
             class_dict["__doc__"] += _EXAMPLE_TEMPLATE % dict(path=path)
 
-        return type.__new__(meta_cls, class_name, bases, class_dict)
+        return super(MetaHasProps, meta_cls).__new__(meta_cls, class_name, bases, class_dict)
 
     def __init__(cls, class_name, bases, nmspc):
         if class_name == 'HasProps':
@@ -555,8 +550,7 @@ def abstract(cls):
 
     return cls
 
-@add_metaclass(MetaHasProps)
-class HasProps(object):
+class HasProps(with_metaclass(MetaHasProps, object)):
 
     def __init__(self, **properties):
         super(HasProps, self).__init__()
```